Zend/zend_object_handlers.c

```diff
@@ -874,6 +880,9 @@ ZEND_API zval *zend_std_write_property(zend_object *zobj, zend_string *name, zva
 				variable_ptr = &EG(error_zval);
 				goto exit;
 			}
+			if (EXPECTED(!(zobj->ce->ce_flags & ZEND_ACC_ALLOW_DYNAMIC_PROPERTIES))) {
```

Comment:
nit: Should this be UNEXPECTED or EXPECTED? I would have thought most uses of dynamic properties would be stdClass or userland classes adding this to avoid notices (to avoid deprecation notices in syslogs or phpunit runs)? This method is `ZEND_COLD`

If emitting a notice is EXPECTED, document why?